disnake/welcome_screen.py

```diff
@@ -0,0 +1,173 @@
+"""
+The MIT License (MIT)
+
+Copyright (c) 2021-present Disnake Development
+
+Permission is hereby granted, free of charge, to any person obtaining a
+copy of this software and associated documentation files (the "Software"),
+to deal in the Software without restriction, including without limitation
+the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the
+Software is furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+DEALINGS IN THE SOFTWARE.
+"""
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, List, Optional
+
+from .partial_emoji import PartialEmoji
+from .utils import MISSING, _get_as_snowflake
+
+if TYPE_CHECKING:
+    from .guild import Guild
+    from .state import ConnectionState
+    from .types.welcome_screen import (
+        WelcomeScreen as WelcomeScreenPayload,
+        WelcomeScreenChannel as WelcomeScreenChannelPayload,
+    )
+__all__ = ("WelcomeScreen", "WelcomeScreenChannel")
+
+
+class WelcomeScreenChannel:
+    """Represents a Discord welcome screen channel.
+
+    .. versionadded:: 2.4
+
+    Attributes
+    ----------
+    id: :class:`int`
+        The ID of the guild channel this welcome screen channel represents.
+    description: :class:`str`
+        The description of this channel in the official UI.
+    emoji: Optional[:class:`PartialEmoji`]
+        The emoji associated with this channel's welcome message, if any.
+    """
+
+    def __init__(self, *, id: int, description: str, emoji: Optional[PartialEmoji] = None):
```

Comment:
I think we should support `str` for emojis as well. This way it will be easier to use unicode emojis and hardcoded custom emojis. Another question is what to do with the state but I think that people won't need it if the object is created on their side.

Reply:
This should be added now